lib/chunks.js

```diff
@@ -525,6 +535,45 @@ module.exports.diffChunks = async ({ coursePath, courseId, courseData, changedFi
     }
   );
 
+  // Check for any deleted course instances or their assessments.
+  await Promise.all(
+    Object.entries(existingCourseChunks.courseInstances).map(async ([ciid, courseInstanceInfo]) => {
+      const courseInstanceExists = !!courseData.courseInstances[ciid];
+      const clientFilesCourseInstanceExists = await fs.pathExists(
```

Comment:
Note that for both `clientFilesCourseInstance` and `clientFilesAssessment`, we need to actually hit the disk, as we can't tell from `courseData` whether or not these directories exist. I expect this to result in a very minor increase in sync time, but it should be dwarfed by the time spend actually doing the sync itself.

Reply:
Makes sense.